metricbeat/docs/fields.asciidoc

```diff
@@ -30829,6 +30847,146 @@ type: long
 Number of blocks on the device that are in sync, in 1024-byte blocks.
 
 
+type: long
+
+--
+
+[float]
+=== service
+
+metrics for system services
+
+
+
+*`system.service.name`*::
++
+--
+The name of the service
+
+type: keyword
+
+--
+
+*`system.service.load_state`*::
++
+--
+The load state of the service
+
+type: keyword
+
+--
+
+*`system.service.state`*::
++
+--
+The activity state of the service
+
+type: keyword
+
+--
+
+*`system.service.sub_state`*::
++
+--
+The sub-state of the service
+
+type: keyword
+
+--
+
+*`system.service.state_since`*::
++
+--
+The timestamp of the last state change. If the service is active and running, this is its uptime.
+
+type: date
+
+--
+
+*`system.service.exec_code`*::
++
+--
+The SIGCHLD code from the service's main process
+
+type: keyword
+
+--
+
+[float]
+=== resources
+
+system metrics associated with the service
+
+
+*`system.service.resources.cpu.usage.ns`*::
++
+--
+CPU usage in nanoseconds
+
+type: long
+
+--
+
+*`system.service.resources.memory.usage.bytes`*::
++
+--
+memory usage in bytes
+
+type: long
+
+--
+
+*`system.service.resources.tasks.count`*::
++
+--
+number of tasks associated with the service
+
+type: long
+
+--
+
+[float]
+=== network
+
+network resource usage
+
+
+*`system.service.resources.network.in.bytes`*::
++
+--
+bytes in
+
+type: long
+
+format: bytes
+
+--
+
+*`system.service.resources.network.in.packets`*::
++
+--
+packets in
+
+type: long
+
+format: bytes
+
+--
+
+*`system.service.resources.network.out.packets`*::
++
+--
+packets out
+
+type: long
+
+--
+
+*`system.service.resources.network.out.bytes`*::
```

Comment:
There are also ECS fields for networking fields, but not sure if we should use `source.bytes` or `server.bytes` here, and we may also consider storing the sum in `network.bytes` (and same for packets).

@webmat do you think that ECS fields should be used here for these networking metrics, and which ones should we use in this case where the direction is not known?

Reply:
I'm not really sure how many of those make sense? We're dealing only with packets from a particular process (or well, groups of processes) and not a whole  network connection, the naming of `network` and `server` seems a bit incongruous.

Reply:
Yeah I agree with @jsoriano on *not* using source.bytes nor server.bytes, these really make semantic sense only for network events, not so much for reporting metrics about an entity.

I would however suggest using `network.bytes` and `network.packets` for the in+out sums. I think the `network.*` field set works as well for entity-based metrics as it does for reporting on network events.

I also support using custom fields for in/out metrics for now, as ECS doesn't have anything that works well for that. Although the seed has been planted; I think it would be useful to have a place in ECS for entity-based network metrics.